### ISSUE TITLE
DOCTYPE Support

### ISSUE THREAD

**Issue by svedie** [Tickaroo/tikxml#122]
Hello,

does tikxml support DOCTYPE declaration?

For Example:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE EqonomizeDoc>
```

If yes, how can I implement this?

If not, will it be supported or implemented?

Greetings,
svedie

**Comment by svedie** [comment]
Yes, I was not precise enough!

I mean writing the Doctype declaration in the XML file, when I write it back to the file system. Reading of the Doctype declaration is not needed.

**Comment by sockeqwe** [comment]
Yeah, it's more a conceptual problem I think. How do you define doctypes in
tikxml? I mean you mostly annotate classes, how do you annotate Dotype
definitions? One could add them in TikXml builder globaly, but that would
mean Doctypes are written always whenever you are using this instance of
tikxml. The real problem is that TikXml misses the concept of a a document.

svedie <notifications@github.com> schrieb am So., 13. Jan. 2019, 11:40:

**Comment by svedie** [comment]
And if the TikXml.Builder has additional method like addTypeAdapter() where the user can add additional header information and this can be what ever he wants?

Therefore if this methode is used, it will be written down, if not, no one is affected.

**Comment by sockeqwe** [comment]
That would work, however, the consequences are that this doctype is written for all xml output from that TikXml instance. This means if you would like to write XML you always write the doctypes. You can't say on the document X write DocType A, on document Y write DocType B and on document Z don't write any doctype. That might be a problem. Workaround would be to have a TikXml instance for Documents who should contain DocType A, a TikXml instance for Documents who want to contain DocType B and a TikXml instance for Document without doctype definitions. This can escalate quickly. Let me check how other xml parsers deal with that ...

**Comment by svedie** [comment]
Thank you for your review.

If the effort is greater than the result, you can close this issue. The resources could be better spend on other features or bugs.

**Comment by reline** [comment]
@sockeqwe I'd actually be interested in implementing this feature, I'll be doing it regardless in a fork for my use case [here](https://github.com/reline/Jisho/blob/701c2e0cd48f4d5ced64b0b86e16f22d86e982f3/hello.xml) but it would be nice to have some direction to see if I can eventually merge it in. Thoughts?
